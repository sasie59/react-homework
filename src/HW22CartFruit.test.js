import React from 'react';
import { render,cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW22CartFruit';

afterEach(cleanup);

test('should HW22CartFruit', () => {
  const onSubmit = jest.fn(
    (index,value) => ({index, value})
  );
  const {container } = render(
    <HW
      name='蘋果'
      price={20}
      index={1}
      onSubmit={onSubmit}
    />);
  const input = container.querySelector('input');
  userEvent.type(input, '25{enter}');
  expect(input.textContent).toBe('');
  expect(container).toMatchSnapshot();
});
