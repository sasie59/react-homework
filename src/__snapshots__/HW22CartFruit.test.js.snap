// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW22CartFruit 1`] = `
<div>
  <div>
    <span>
      蘋果
    </span>
    <span>
      $
      20
    </span>
    <form>
      <input
        name="蘋果"
        placeholder="數量"
        type="number"
        value="0"
      />
      <button>
        加入購物車
      </button>
    </form>
  </div>
</div>
`;
